### PR TITLE
feat(grouping): Fewer sentinels

### PR DESCRIPTION
**Pull request by jjbayer**
Do not mark `malloc`, `free`, and `throw` categories as sentinel.
Skip prefix frames when looking for blaming frame in fallback tree.

https://getsentry.atlassian.net/browse/INGEST-144

### CODE REVIEW

File: tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/mobile@2021_02_12/group_432_event_432.pysnap [hunk 1]
```diff
@@ -1,19 +1,124 @@
...
           frame*
             function*
               "stripped_application_code"
-          frame* (marked as prefix frame by stack trace rule (category:throw +sentinel +prefix ^-group))
+        type (ignored because exception is synthetic)
+          "0x40000015 / 0x00000001"
+        value (ignored because stacktrace takes precedence)
+          "Fatal Error: <hex> / <hex>"
+--------------------------------------------------------------------------
+app-depth-2:
```
Comment (jjbayer) [inline tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/mobile@2021_02_12/group_432_event_432.pysnap:21]: @untitaker Are all these extra levels OK?

**Comment by jjbayer** [comment]
Snapshot changes look good to me.

File: tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/mobile@2021_02_12/group_432_event_432.pysnap (hunk 1 above)
Comment (untitaker) [reply tests/sentry/grouping/snapshots/test_variants/test_event_hash_variant/mobile@2021_02_12/group_432_event_432.pysnap:21]: yeah that's okay, this now drops into fallback mode

File: tests/sentry/grouping/test_categorization.py [hunk 2]
```diff
@@ -59,7 +59,9 @@
 _fixture_path = os.path.join(os.path.dirname(__file__), "categorization_inputs")
 
 _SHOULD_DELETE_DATA = os.environ.get("SENTRY_TEST_GROUPING_DELETE_USELESS_DATA") == "1"
-_DELETE_KEYWORDS = os.environ.get("SENTRY_TEST_GROUPING_DELETE_KEYWORDS", "").lower().split(",")
+_DELETE_KEYWORDS = [
+    x for x in os.environ.get("SENTRY_TEST_GROUPING_DELETE_KEYWORDS", "").lower().split(",") if x
```
Comment (untitaker) [inline tests/sentry/grouping/test_categorization.py:63]: for context, this fixes a bug where all snapshots were deleted on machines that did not have this environment variable set

**Review by untitaker — APPROVED** [review]

**Pull request merged**